scipy/linalg/tests/test_decomp_polar.py

```diff
@@ -91,3 +92,29 @@ def test_verify_cases():
     for a in verify_cases:
         verify_polar(a)
 
+@pytest.mark.parametrize('dt', [int, float, np.float32, complex, np.complex64])
+def test_empty(dt):
+    empty_cases = [
+    np.empty((0, 0), dt),
+    np.empty((0, 2), dt),
+    np.empty((2, 0), dt),
+    ]
+
+    for a in empty_cases:
+        # a = up 
+        u, p = polar(a, side='right')
+        u_n, p_n = polar(np.eye(2, dtype=dt))
+
+        assert_equal(u.dtype, u_n.dtype)
+        assert_equal(p.dtype, p_n.dtype)
+        assert_(u.size == 0)
```

Comment:
It is stronger to test the `shape`, which depends on `side='right'` vs `side='left`. I'll go ahead and do that.